File: greedy/1400_construct_k_palindrome_strings/1400_construct_k_palindrome_strings.py

class Solution:
    def canConstruct(self, s: str, k: int) -> bool:
        digit_count = {}
        if len(s) < k:
            return False
        elif len(s) == k:
            return True
        else:
            odd = 0
            for i in set(s):
                digit_count[i] = s.count(i)
            
            for i in digit_count.values():
                if i % 2 != 0:
                    odd += 1
            if odd > k:
                return False
            else:
                return True


s = "annabelle"
k = 2
res = Solution().canConstruct(s, k)
print(res)

File: greedy/1400_construct_k_palindrome_strings/1400_construct_k_palindrome_strings.go
package main

import "fmt"

func canConstruct(s string, k int) bool {
	n := len(s)
	if n < k {
		return false
	}
	if n == k {
		return true
	}

	cnt := make([]int, 26)

	for i := 0; i < n; i++ {
		cnt[int(s[i]-'a')]++
	}

	var odd int

	for i := 0; i < 26; i++ {
		odd += cnt[i] & 1
	}

	return odd <= k
}

func main() {
	s := "annabelle"
	k := 2
	res := canConstruct(s, k)
	fmt.Println(res)
}
